agent/api/task_windows.go

```diff
@@ -50,3 +53,14 @@ func (task *Task) downcaseAllVolumePaths() {
 func getCanonicalPath(path string) string {
 	return filepath.Clean(strings.ToLower(path))
 }
+
+// platformHostConfigOverride Sets up default HostConfig options to be passed to Docker API.
+// Docker for Windows treats some options different than Unix and some are not supported at all.
+// Version 1.12.x of Docker for Windows would ignore the unsupported option MemorySwappiness.
+// Version 17.03.x will cause an error if any value other than -1 is passed in for MemorySwappiness.
+// This bug is not noticed when no value is passed in. However, the go-dockerclient client version
+// we are using removed the json option omitempty causing this parameter to default to 0 if empty.
+// https://github.com/fsouza/go-dockerclient/commit/72342f96fabfa614a94b6ca57d987eccb8a836bf
+func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) {
+	hostConfig.MemorySwappiness = memorySwappinessDefault
```

Comment:
Would you recommend returning a new copy that has been modified, or moving the method to be attached to hostConfig (i.e. hostConfig.platformOverride())